### DISCUSSION TITLE
No such organization

### DISCUSSION THREAD

**Discussion by JudicaMeAndrzej** [srcnalt/OpenAI-Unity#67] [Q&A]
Hello!

I did not see any thread about this in the forums here, so I apologize if this has been addressed elsewhere. 

I will need users to pass in their API keys into the game (I have no idea what best practice here but I certainly can't just allow an unknown number of players to make an unpredictable amount of API calls using my personal key). However, when I try call new OpenAIAPI(...), passing my key and organization, when I make a request, I get the error "Error Message: No such organization: " and my org key. 

Note that it works fine if it is using no parameters, that is, looking for my auth.json.

How can I get around this issue?

Thanks greatly,
Andrew

**Comment by srcnalt** [comment]
Hi Andrew,

If the message is coming from OpenAI then it does not recognize the org id. The package does not process the keys, just sends them to OpenAI in the headers. I'd suggest checking if there are any white spaces in the key.

Also just released #68 the issue might be JSON parsing issue which should be aided with latest release v0.1.14
